windows/devices/uwb/UwbCxAdapterDdiLrp.cxx

```diff
@@ -1635,15 +1623,70 @@ windows::devices::uwb::ddi::lrp::To(const UWB_APP_CONFIG_PARAM &applicationConfi
     return uwbApplicationConfigurationParameter;
 }
 
+UwbApplicationConfigurationParameter
+windows::devices::uwb::ddi::lrp::To(const UWB_APP_CONFIG_PARAM &applicationConfigurationParameter, const ::uwb::UwbMacAddressType macAddressMode)
+{
+    UwbApplicationConfigurationParameter uwbApplicationConfigurationParameter{
+        .Type = To(applicationConfigurationParameter.paramType)
+    };
+
+    if (applicationConfigurationParameter.paramType == UWB_APP_CONFIG_PARAM_TYPE_DST_MAC_ADDRESS) {
+        std::unordered_set<::uwb::UwbMacAddress> value{};
+
+        if (macAddressMode == ::uwb::UwbMacAddressType::Short) {
+            for (auto i = 0; i < (applicationConfigurationParameter.paramLength / ::uwb::UwbMacAddress::ShortLength); i++) {
+                ::uwb::UwbMacAddress::ShortType data{};
+                auto offset = i * ::uwb::UwbMacAddress::ShortLength;
+                std::memcpy(std::data(data), &applicationConfigurationParameter.paramValue[0] + offset, std::size(data));
+                value.insert(::uwb::UwbMacAddress(std::move(data)));
+            }
+        } else { // ::uwb::UwbMacAddressType::Extended
+            for (auto i = 0; i < (applicationConfigurationParameter.paramLength / ::uwb::UwbMacAddress::ExtendedLength); i++) {
+                ::uwb::UwbMacAddress::ExtendedType data{};
+                auto offset = i * ::uwb::UwbMacAddress::ExtendedLength;
+                std::memcpy(std::data(data), &applicationConfigurationParameter.paramValue[0] + offset, std::size(data));
+                value.insert(::uwb::UwbMacAddress(std::move(data)));
+            }
+        }
+
+        uwbApplicationConfigurationParameter.Value = std::move(value);
+    } else {
+        PLOG_ERROR << "unexpected uwb application configuration parameter type encountered, type=" << std::showbase << std::hex << notstd::to_underlying(applicationConfigurationParameter.paramType);
+    }

```

Comment:
This should throw a `std::runtime::error` because this would be a programmer bug (ie. incorrect use of function). So, we'll want to identify such bugs before we commit the code.

Reply:
Good idea, I'll add it